Compiling usbd.h with C++ compiler fails

Hi,

Compiling a Zephyr project with a C++ compiler (CONFIG_CPP=y) leads to a build error, when `zephyr/include/zephyr/usb/usbd.h` gets included.

```
/opt/zephyrproject/zephyr/include/zephyr/usb/usbd.h:507:9: error: designator order for field 'usbd_desc_node::<anonymous>' does not match declaration order in 'usbd_desc_node'
  507 |         }
```

```
/opt/zephyrproject/zephyr/include/zephyr/usb/usbd.h:530:9: error: designator order for field 'usbd_desc_node::ptr' does not match declaration order in 'usbd_desc_node'
  530 |         }
```

Putting the members of the struct in the correct order in those macros fixes the problem.

--- a/include/zephyr/usb/usbd.h
+++ b/include/zephyr/usb/usbd.h
@@ -497,13 +497,13 @@ static inline void *usbd_class_get_private(const struct usbd_class_data *const c
 #define USBD_DESC_LANG_DEFINE(name)					\
 	static uint16_t langid_##name = sys_cpu_to_le16(0x0409);	\
 	static struct usbd_desc_node name = {				\
-		.bLength = sizeof(struct usb_string_descriptor),	\
-		.bDescriptorType = USB_DESC_STRING,			\
 		.str = {						\
 			.idx = 0,					\
 			.utype = USBD_DUT_STRING_LANG,			\
 		},							\
 		.ptr = &langid_##name,					\
+		.bLength = sizeof(struct usb_string_descriptor),	\
+		.bDescriptorType = USB_DESC_STRING,			\
 	}
 
 /**
@@ -524,9 +524,9 @@ static inline void *usbd_class_get_private(const struct usbd_class_data *const c
 			.utype = d_utype,					\
 			.ascii7 = true,						\
 		},								\
+		.ptr = &ascii_##d_name,						\
 		.bLength = USB_STRING_DESCRIPTOR_LENGTH(d_string),		\
 		.bDescriptorType = USB_DESC_STRING,				\
-		.ptr = &ascii_##d_name,					\
 	}
 
 /**

